crates/consensus/src/header.rs

```diff
@@ -638,6 +638,158 @@ impl<'a> arbitrary::Arbitrary<'a> for Header {
     }
 }
 
+/// Trait for extracting specific Ethereum block data from a header
+pub trait BlockHeader {
```

Comment:
from reth pov there's an argument for limiting these functions to the absolute necessary fields, or introduce sub traits, for example for evm you only need certain fields.
but since this doesn't clash with rpc, I think all of those are fine, and we could still do sub traits

Reply:
Let me know if you think we should split in subtraits now or if we should do that in a follow up if needed

Reply:
I think we can do in a followup if needed, realistically we only need this trait for one type really, the chain's header type and for op we can even reuse the L1 type